enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,522 @@
+components will be listed in the [ClusterVersion](https://github.com/openshift/api/blob/4436dc8be01e8dcd8b250e1b32bb0fbd64ba78ac/config/v1/types_cluster_version.go#L35) object:
+
+```yaml
+  spec:
+    capabilities:
+      inclusionDefault: [Include|Exclude]
+      exclude:
+      - console
+      - samples
+      include:
+      - registry
+```
+
+The CVO will filter out(not apply/reconcile) resources that are annotated with key/value
+`capability.openshift.io/name=$exclusions` and listed in the `exclude` section, or are not listed
+and the `default` setting is `exclude`.  In addition any explicitly included capability
+should be added, regardless of the [CLUSTER_PROFILE](https://github.com/openshift/enhancements/blob/7da960b98b2835022823f569efa1e18e9e410696/enhancements/update/cluster-profiles.md).
+
+If a resource has multiple `capability.openshift.io/name` annotations, then the resource will be filtered if any
+of the annotations on the resource are matched by a configured filter.  This allows a component to define
+a `ServiceMonitor` which is part of both the `Monitoring` capability and the `Foo` capability.  If either
+of those capabilities are excluded, the `ServiceMonitor` should not be created.  (If Monitoring is excluded
+then no ServiceMonitors should be created, if only Foo is excluded, then only Foo's ServiceMonitors should
+be excluded).
+
+Specifying the same capability as an `include` and an `exclude` is an error condition.  This avoids the need 
+to define behavior for what to do when a user specifies the same capability in both, and update a status field
+indicating what choice the CVO made.  To enforce this the CVO will need to evaluate the spec values and, if
+they are valid, copy them into status.  If they are not valid, status must contain an indication of what is
+invalid and why.  This also means the CVO must be driven off the status values, not the spec values.
+
+Note: The CVO must wait until the ClusterVersion resource exists (created by the installer) before creating
+any filterable resource(a resource with a capability annotation) to ensure there is no race condition in which
+the CVO starts creating resources that should have been filtered, before it has the filter list.
+
+4) Admin can remove an item from the excluded annotations list, but they cannot add an item to it.  If an
+item is removed, the CVO will apply the previously filtered(skipped) resources to the cluster on the next reconciliation.
+Adding an item to the filtered list is not supported because it requires the component be removed from the
+running cluster which has more significant implications for how all traces of the component are removed.  The
+CVO should reject admin edits/updates that add new entries to the exclusion list.  Again this requires the
+introduction of a validating webhook for the ClusterVersion resource.
+
+Similarly, changing the `inclusionDefault` from `include` to `exclude` must be prevented.
```

Comment:
s/must be prevented/ is not possible after the installation of the cluster has completed/

Reply:
will rephrase